src/stirling/source_connectors/socket_tracer/bcc_bpf/socket_trace.c

```diff
@@ -585,6 +585,57 @@ int conn_cleanup_uprobe(struct pt_regs* ctx) {
   return 0;
 }
 
+// These probes are used to capture the *sock struct during client side tracing
+// of connect() syscalls. This is necessary to capture the socket's local address,
+// which is not accessible via the connect() and later syscalls.
+//
+// This function requires that the function being probed receives a struct sock* as its
+// first argument and that the active_connect_args_map is populated when this probe fires.
+// This means the function being probed must be part of the connect() syscall path or similar
+// syscall path.
+//
+// Using the struct sock* for capturing a socket's local address only works for TCP sockets.
+// The equivalent UDP functions (udp_v4_connect, udp_v6_connect and upd_sendmsg) always receive a
+// sock struct with a 0.0.0.0 or ::1 local address. This is deemed acceptable since our local
+// address population for server side tracing relies on accept/accept4, which only applies for TCP.
+//
+// TODO(ddelnano): The current implementation works for mid stream TCP connections despite
+// my intuition that tcp_v4_connect and tcp_v6_connect should not be called mid stream.
+// If mid stream connections have a missing local address in the future, we should probe
+// tcp_sendmsg in addition to the current probes. This will require discerning between the
+// UDP and TCP case in process_implicit_conn.
```

Comment:
I tried to test this mid stream case by creating a psql shell prior to the PEM starting and issuing queries after the socket tracer was initialized. I verified that a tcp connection was made and only `sendto` syscalls were issued when queries were executed (no `connect` syscalls) to check that this would simulate a mid stream case.

Surprisingly, I saw that the `local_addr` columns were populated despite my thinking that it should be missed. I intend to add a small addition to this change to cover the mid stream case even though I can't seem to reproduce it failing. This would work by grabbing the `sock*` struct from one additional kprobe `tcp_sendmsg` and handling it in the process_implict_conn. The slight complication here is that the UDP case mentioned above needs to be filtered out. 

I wanted to get feedback on the current scope of this before doing that part and to see there were ideas on why the testing above seemed to work.